modules/olm-bundle/01_mod.mk

```diff
@@ -0,0 +1,125 @@
+current_makefile = $(lastword $(MAKEFILE_LIST))
+current_makefile_directory = $(dir $(current_makefile))
+yamlfmt_config := $(current_makefile_directory)/yamlfmt.yaml
+comma := ,
+
+base64_wrapped := base64 -w76
+ifeq ($(HOST_OS),darwin)
+	base64_wrapped := base64 -b76
+endif
+
+# Utility functions
+for_each_kv = $(foreach item,$2,$(eval $(call $1,$(word 1,$(subst =, ,$(item))),$(word 2,$(subst =, ,$(item))))))
+fatal_if_undefined = $(if $(findstring undefined,$(origin $1)),$(error $1 is not set))
+
+# Build up config
+olm_bundle_dir := $(bin_dir)/scratch/olm/bundle-$(oci_olm_image_tag)
+
+# Build the olm-to-oci tool
+olm_to_oci_dir := $(current_makefile_directory:/=)/olm-to-oci
+OLM-TO-OCI := $(CURDIR)/$(bin_dir)/tools/olm-to-oci
+NEEDS_OLM-TO-OCI := $(bin_dir)/tools/olm-to-oci
+$(NEEDS_OLM-TO-OCI): $(wildcard $(olm_to_oci_dir)/*.go) | $(NEEDS_GO)
+	cd $(olm_to_oci_dir) && GOWORK=off GOBIN=$(CURDIR)/$(dir $@) $(GO) install .
+
+# Get all example files
+olm_examples := $(wildcard $(olm_example_directory)/*.yaml)
+
+.PHONY: olm-bundle
+olm-bundle: $(helm_chart_archive) $(olm_clusterserviceversion_path) | $(NEEDS_HELM) $(NEEDS_YQ) $(NEEDS_OPERATOR-SDK) $(NEEDS_YAMLFMT)
+	rm -rf $(olm_bundle_dir) $(oci_layout_path_olm) $(oci_digest_path_olm)
+	mkdir -p $(olm_bundle_dir)
+	cd $(olm_bundle_dir) && $(HELM) template $(addprefix --values=,$(abspath $(olm_helm_value_path))) $(olm_project_name) $(CURDIR)/$(helm_chart_archive) |\
+		$(YQ) 'del(.metadata.annotations["helm.sh/resource-policy"])' |\
+		$(YQ) 'del(.metadata.labels["app.kubernetes.io/managed-by"])' |\
+		$(YQ) 'del(.metadata.labels["app.kubernetes.io/version"])' |\
+		$(YQ) 'del(.metadata.labels["helm.sh/chart"])' |\
+		$(YQ) 'del(.metadata.creationTimestamp)' |\
+		$(YQ) - $(abspath $(olm_clusterserviceversion_path)) $(abspath $(olm_examples)) $(abspath $(olm_additional_manifests)) |\
+		$(OPERATOR-SDK) generate bundle --output-dir . --package $(olm_project_name) --version $(VERSION:v%=%)
+	
+	@# Set the container image annodation
+	$(YQ) -i '.metadata.annotations.containerImage = .spec.install.spec.deployments[0].spec.template.spec.containers[0].image' $(olm_bundle_dir)/manifests/$(olm_project_name).clusterserviceversion.yaml 
+	$(YQ) -i '.spec.relatedImages = [ .spec.install.spec.deployments[].spec.template.spec.containers[] | {"name": .name, "image": .image} ]' $(olm_bundle_dir)/manifests/$(olm_project_name).clusterserviceversion.yaml 
+
+	@# Set the supported OS/Arches based on $(oci_platforms)
+	$(YQ) -i --string-interpolation '. * ("$(oci_platforms)" | [ split(",").[] | split("/") | ["operatorframework.io/os.\(.[0])", "operatorframework.io/arch.\(.[1])"] ] | [ .[][] ] | unique | .[] as $$item ireduce({}; . * {"metadata": {"annotations": {$$item: "supported"}}}))' $(olm_bundle_dir)/manifests/$(olm_project_name).clusterserviceversion.yaml 
+	
+	@# Set the openshift version
+	$(YQ) -i '.annotations."com.redhat.openshift.versions"="v4.6"' $(olm_bundle_dir)/metadata/annotations.yaml
```

Comment:
Do we place this version in a variable with a comment to explain what v4.6 means and when we should update it?